synapse/util/threepids.py

```diff
@@ -18,6 +18,16 @@
 logger = logging.getLogger(__name__)
 
 
+# it's unclear what the maximum length of an email address is. RFC3696 (as corrected
+# by errata) says:
+#    the upper limit on address lengths should normally be considered to be 254.
+#
+# In practice, mail servers appear to be more tolerant and allow 400 characters
+# or so. Let's allow 500, which should be plenty for everyone.
```

Comment:
Is this for the full email address (including the domain name)?

Reply:
FWIW I stole this from sydent https://github.com/matrix-org/sydent/blob/c2ab10897500dbba1c0d4be33f42985d81bbad7d/sydent/util/stringutils.py#L29-L36

I don't know if it includes a domain name, but also :shrug: